Fix log export ordering

takenagain opened the pull request:
- Add zero-padding to day and month in the filename 
- Add sorting to the `exportLogsStream` functions

takenagain committed:
Add padding to the month and day in the output file name

takenagain committed:
Add log export unit tests

takenagain committed:
Fix file order in file export for web and native

takenagain committed:
Export logs to file in test for visual confirmation

takenagain committed:
Clean log files after test with `tearDown` function

CharlVS reviewed (commented):
Thank you for your contribution. Please run `dart fix --apply` to resolve the following issues:

```
* line 2, column 1 of test/mock_path_provider_platform.dart: This package does not have path_provider_platform_interface in the `dependencies` or `dev_dependencies` section of `pubspec.yaml`.
    ╷
  2 │ import 'package:path_provider_platform_interface/path_provider_platform_interface.dart';
    │ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ╵
* line 3, column 1 of test/mock_path_provider_platform.dart: This package does not have plugin_platform_interface in the `dependencies` or `dev_dependencies` section of `pubspec.yaml`.
    ╷
  3 │ import 'package:plugin_platform_interface/plugin_platform_interface.dart';
    │ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ╵
* line 8, column 1 of test/dragon_logs_test.dart: This package does not have path_provider_platform_interface in the `dependencies` or `dev_dependencies` section of `pubspec.yaml`.
    ╷
  8 │ import 'package:path_provider_platform_interface/path_provider_platform_interface.dart';
    │ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ╵
```

takenagain committed:
Fix flutter analyze warnings

Using Flutter `3.16.9` to stick with the version used when initially creating the PR

takenagain committed:
Update dependencies to Flutter 3.22
